src/vec_runtime_DYN.c

```diff
@@ -0,0 +1,210 @@
+ *
+ * The working assumption the base and highest supported platform
+ * (-mcpu=) for Big/Little Endian can and should be different.
+ *
+ * The base platform for Big Endian is POWER7 as the first
+ * implementation of Vector Scalar Extended (VXS). The highest
+ * supported platform is POWER8.
+ * Distros supporting Big Endian are older and usual restricted to
+ * POWER8 Compatibility mode when running on POWER9 hardware.
+ *
+ * The base platform for Little Endian is POWER8 as the first
+ * officially supported LE platform. The highest supported platform
+ * is is currently POWER9.  All the latest Distros are Little Endian.
+ *
+ * We depend on the compiler and runtime (GLIBC 2.23) to support
+ * __builtin_cpu_is() in the resolver function.
+ * GCC support for __builtin_cpu_is() on PPC started
+ * with version 6. You need both (GCC 6 or later and GLIBC 2.23 or
+ * later).  We check that the compiler defines the macro
+ * __BUILTIN_CPU_SUPPORTS__ to verify support is available.
+ *
+ * If __BUILTIN_CPU_SUPPORTS__ is not defined we default to the
+ * appropriate (for the platform endian) base platform.
+ */
+
+#include <pveclib/vec_int512_ppc.h>
+
+/*! \brief Macro to expand the parameterize resolver. */
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#ifdef  __BUILTIN_CPU_SUPPORTS__
+#define VEC_DYN_RESOLVER(FNAME) \
+  if (__builtin_cpu_is ("power8")) \
+    return FNAME ## _PWR8; \
+  else \
+    return FNAME ## _PWR7;
+#else
+#define VEC_DYN_RESOLVER(__FNAME)
+   return FNAME ## _PWR7;
+#endif
+#else // __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#ifdef  __BUILTIN_CPU_SUPPORTS__
+#define VEC_DYN_RESOLVER(FNAME) \
+  if (__builtin_cpu_is ("power9")) \
+    return FNAME ## _PWR9; \
+  else \
+    return FNAME ## _PWR8;
+#else
+#define VEC_DYN_RESOLVER(FNAME) \
+   return FNAME ## _PWR8;
+#endif
+#endif
+
+
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+/* Not BIG Endian, lets call the whole thing off.
+   LITTLE Endian requires power8 or later.  */
+
+extern __VEC_U_256
+vec_mul128x128_PWR7 (vui128_t, vui128_t);
+
+extern __VEC_U_512
+vec_mul256x256_PWR7 (__VEC_U_256, __VEC_U_256);
+
+extern __VEC_U_640
+vec_mul512x128_PWR7 (__VEC_U_512, vui128_t);
+
+extern __VEC_U_1024
+vec_mul512x512_PWR7 (__VEC_U_512, __VEC_U_512);
+
+extern void
+vec_mul1024x1024_PWR7 (__VEC_U_2048 *, __VEC_U_1024 *, __VEC_U_1024 *);
+
+extern void
+vec_mul2048x2048_PWR7 (__VEC_U_4096 *, __VEC_U_2048 *, __VEC_U_2048 *);
+#endif
+
+extern __VEC_U_256
+vec_mul128x128_PWR8 (vui128_t, vui128_t);
+
+extern __VEC_U_512
+vec_mul256x256_PWR8 (__VEC_U_256, __VEC_U_256);
+
+extern __VEC_U_640
+vec_mul512x128_PWR8 (__VEC_U_512, vui128_t);
+
+extern __VEC_U_1024
+vec_mul512x512_PWR8 (__VEC_U_512, __VEC_U_512);
+
+extern void
+vec_mul1024x1024_PWR8 (__VEC_U_2048 *, __VEC_U_1024 *, __VEC_U_1024 *);
+
+extern void
+vec_mul2048x2048_PWR8 (__VEC_U_4096 *, __VEC_U_2048 *, __VEC_U_2048 *);
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+/* Not LITTLE Endian, lets call the whole thing off.
```

Comment:
This comment seems out of place.

Reply:
Ok 

/* Older distros running Big Endian are unlikely to support PWR9.
 * So declare PWR9 externs only for LE.  */